frontend/src/Components/Elements/Post.tsx

```diff
@@ -26,12 +28,23 @@ function Post({
 }: PostProps) {
   return (
     <div className="timeline-box flex flex-col overflow-clip">
+      {/* Add conditional rendering for UsernameRepost */}
+      <div className="-mx-3 mb-4 flex flex-row justify-end border-b border-gray-500 px-6 pb-1">
+        <UsernameRepost username={username} />
+      </div>
+
       <div className="flex flex-row items-center gap-4">
         <ProfilePicture width={80} image={profileImage} />
         <h5>{profileName}</h5>
         <p className="text-black50">{username}</p>
         <p className="ml-auto mr-3 self-start">{time.toLocaleString()}</p>
       </div>
+
+      {/* Add conditional rendering for InReplyTo */}
+      <div className="-mx-3 mt-4 flex flex-row justify-start border-y border-gray-500 px-4 py-4">
```

Comment:
```suggestion
      <div className="-mx-3 mt-4 flex flex-row justify-start border-y border-black25 px-4 py-4 dark:border-white25">
```
Same as above. Also the dark mode because the colors are dimmer now (based them on the figma doc).